### PR TITLE
HHH-18414 Remove duplicated call of `callAttributeBindersInSecondPass(property)`

### PR DESCRIPTION
**Pull request by quaff**
Prior to this commit, `AttributeBinder` will be called twice, it will produce duplicated tenant filter.

----------------------
By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt)
and can be relicensed under the terms of the [LGPL v2.1 license](https://www.gnu.org/licenses/old-licenses/lgpl-2.1.txt) in the future at the maintainers' discretion.
For more information on licensing, please check [here](https://github.com/hibernate/hibernate-orm/blob/main/CONTRIBUTING.md#legal).

----------------------

https://hibernate.atlassian.net/browse/HHH-18414

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/mapping/PersistentClass.java [hunk 1]
```diff
@@ -876,6 +876,11 @@ public void addFilter(
 			boolean autoAliasInjection,
 			java.util.Map<String, String> aliasTableMap,
 			java.util.Map<String, String> aliasEntityMap) {
+		for ( FilterConfiguration existing : filters ) {
+			if ( existing.getName().equals( name ) && existing.getCondition().equals( condition ) ) {
```
Comment (quaff) [inline hibernate-core/src/main/java/org/hibernate/mapping/PersistentClass.java:880]: I'm not sure if either `name` or `condition` is enough.

**Comment by quaff** [comment]
@gavinking It seems the regression is introduced by your [commit](https://github.com/hibernate/hibernate-orm/commit/95ee026cb24ef54b8bfb24447ca133ca0f11326f), please review.

**Comment by gavinking** [comment]
I agree with your analysis here, but I don't think the fix is quite perfect. I think it would be better to delete the call to `callAttributeBindersInSecondPass( property )` from `makeProperty()`, and ensure that all callers of `makeProperty()` also call `callAttributeBindersInSecondPass()`, *after* the property is added to its container.

**Comment by gavinking** [comment]
See #8713.

**Comment by quaff** [comment]
> See #8713.

@gavinking Test added.

**Comment by gavinking** [comment]
> > See #8713.
> 
> @gavinking Test added.

Great, thanks, appreciated!

**Pull request closed without merging**